### PR TITLE
docs: cql: split DML page into one page per statement

### PR DESCRIPTION
**Pull request by avikivity**
The DML page is quite long (21 screenfuls on my monitor); split it into one page per statement to make it more digestible.

The sections that are common to multiple statement are kept in the main DML page, and references to them are added.

### CODE REVIEW

File: docs/cql/dml.rst [hunk 1]
```diff
@@ -5,755 +5,13 @@ Data Manipulation
...
-.. _IF:
-
-IF condition
-~~~~~~~~~~~~
-
-It is, however, possible to use the conditions on some columns through ``IF``, in which case the row will not be updated
-unless the conditions are met. Each such ``UPDATE`` gets a globally unique timestamp.
-But, please note that using ``IF`` conditions will incur a non-negligible performance
-cost (internally, Paxos will be used), so this should be used sparingly.
+.. toctree::
```
Comment (tzach) [inline docs/cql/dml.rst:8]:
```suggestion
.. toctree::
   :maxdepth: 1
```

### Before 
image

### After
image

File: docs/cql/dml.rst [hunk 2]
```diff
@@ -5,755 +5,13 @@ Data Manipulation
 
 This section describes the statements supported by CQL to insert, update, delete, and query data.
```
Comment (tzach) [inline docs/cql/dml.rst:6]:
```suggestion
The following pages describe the statements CQL supported to insert, update, delete, and query data, followed by sections on all updates commands.
```

**Review by nyh — APPROVED** [review]
I agree that splitting that page is the right decision, and I like DynamoDB's documentation's decision to have a separate page per instruction. On the other hand, there's a lot of commonality between INSERT, UPDATE and DELETE - and in fact a good explanation of what the heck is the difference between an INSERT and UPDATE seems to be missing (!?). So I feel we should have a page covering those three instructions. But we can always do further reorganizations later.

**Comment by avikivity** [comment]
It's better to have a page per statement, and callouts to common sections. For example, we'll have a shared section about expressions.

**Comment by avikivity** [comment]
v2:
 - limited DML page table of contents to depth 1
 - rephrased introduction to DML page

**Review by annastuchlik — APPROVED** [review]
LGTM. Thanks, @avikivity, this is a nice improvement.

**Comment by avikivity** [comment]
@nyh please re-review

**Pull request closed without merging**